lib/Sema/TypeCheckExpr.cpp

```diff
@@ -709,3 +709,107 @@ bool TypeChecker::isCompatibleWithVectorAutoDiff(Type type, DeclContext *DC) {
   }
   return false;
 }
+
+// SWIFT_ENABLE_TENSORFLOW
+// Returns the function declaration corresponding to the given function name and
+// lookup context. If the function declaration cannot be resolved, emits a
+// diagnostic and returns nullptr.
+FuncDecl *
+TypeChecker::lookupFuncDecl(
+    DeclName funcName, SourceLoc funcNameLoc, DeclContext *lookupContext,
+    const std::function<bool(FuncDecl *)> &isValidFuncDecl,
+    const std::function<void()> &overloadDiagnostic,
+    const std::function<void()> &ambiguousDiagnostic,
+    const std::function<void()> &notFunctionDiagnostic,
+    NameLookupOptions lookupOptions,
+    const Optional<std::function<bool(FuncDecl *)>> &hasValidTypeCtx,
+    const Optional<std::function<void()>> &invalidTypeCtxDiagnostic) {
+
+  FuncDecl *resolvedFuncDecl = nullptr;
+
+  // Perform lookup.
+  auto results =
+    lookupUnqualified(lookupContext, funcName, funcNameLoc, lookupOptions);
+
+  // If looking up an operator within a type context, look specifically within
+  // the type context.
+  // This works around the fact that operators cannot be specified with a
+  // qualified name (i.e. only `(+)` works, `Float.(+)` doesn't).
+  if (funcName.isOperator() && lookupContext->isTypeContext()) {
```

Comment:
How about

```cpp
if (funcName.isOperator() && lookupContext->isTypeContext())
  if (auto tmp = lookupMember(lookupContext, lookupContext->getSelfTypeInContext(), funcName))
    results = tmp;
```

Before writing any `empty()` checks, I'd usually check whether `tmp` has `operator bool()` first. In this case `LookupResult` does have one.

Reply:
Thanks for the tip! Done.